src/core/script.rs

```diff
@@ -1,3 +1,64 @@
+//! Bitcoin script pubkey types and operations.
+//!
+//! This module provides types for working with script pubkeys (also known as
+//! "locking scripts"), which define the conditions that must be met to spend a
+//! [`TxOut`].
+//!
+//! # Overview
+//!
+//! Script Pubkeys specify the spending conditions for outputs in the Bitcoin scripting
+//! language. Common types include:
+//!
+//! - **P2PKH** (Pay-to-Public-Key-Hash): Standard address payments
+//! - **P2SH** (Pay-to-Script-Hash): More complex spending conditions
+//! - **P2WPKH** (Pay-to-Witness-Public-Key-Hash): Native SegWit
+//! - **P2WSH** (Pay-to-Witness-Script-Hash): Native SegWit script hash
+//! - **P2TR** (Pay-to-Taproot): Taproot outputs
```

Comment:
I would remove this common type list. It is not really relevant to how we operate here.